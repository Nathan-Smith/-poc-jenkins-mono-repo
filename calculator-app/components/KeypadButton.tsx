import { useEffect, useRef } from 'react';
import styles from '../styles/Home.module.scss';
import { MDCRipple } from '@material/ripple';

export function KeypadButton({
  gridArea, children,
}: {
  gridArea: string;
  children: React.ReactNode;
}): JSX.Element {
  const buttonRef = useRef<HTMLButtonElement>(null);

  useEffect(() => {
    if (buttonRef.current)
      new MDCRipple(buttonRef.current);
  }, [buttonRef]);

  return (
    <button
      className={`mdc-button mdc-button--outlined ${styles.keypadButton}`}
      style={{ gridArea: gridArea }}
      ref={buttonRef}
    >
      <span className="mdc-button__ripple"></span>
      <span className="mdc-button__label">{children}</span>
    </button>
  );
}
