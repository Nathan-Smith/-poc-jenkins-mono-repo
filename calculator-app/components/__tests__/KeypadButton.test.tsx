import React from 'react'
import renderer, { act, ReactTestRenderer } from 'react-test-renderer'

import { KeypadButton } from '../KeypadButton'

describe('KeypadButton', () => {
  test('renders', async () => {
    let component: ReactTestRenderer | undefined
    act(() => {
      component = renderer.create(<KeypadButton gridArea="a">a</KeypadButton>)
    })

    expect(component?.toJSON()).toMatchSnapshot()
  })
})
