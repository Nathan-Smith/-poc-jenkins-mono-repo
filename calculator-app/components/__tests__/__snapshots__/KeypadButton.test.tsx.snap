// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeypadButton renders 1`] = `
<button
  className="mdc-button mdc-button--outlined keypadButton"
  style={
    Object {
      "gridArea": "a",
    }
  }
>
  <span
    className="mdc-button__ripple"
  />
  <span
    className="mdc-button__label"
  >
    a
  </span>
</button>
`;
